src/APIBasedAuth.ts

```diff
@@ -280,6 +280,8 @@ class APIBasedAuth {
 }
 
 declare namespace APIBasedAuth {
+  type ClientOptions = Required<Config>;
```

Comment:
Does this make `baseUrls.apps` required too?

Reply:
no but should it?

Reply:
right now it would be:
```
{
    clientId: string;
    baseURLs: {
      apps?: string;
      api?: string;
    };
    storage: Storage;
    storageKeys: StorageKeys;
  }
```
but I can make a DeepRequired generic and make it:
```
{
    clientId: string;
    baseURLs: {
      apps: string;
      api: string;
    };
    storage: Storage;
    storageKeys: StorageKeys;
  }
```

by using: 
```
type Primitive = undefined | null | boolean | string | number | Function;
type DeepRequired<T> = T extends Primitive
  ? NonNullable<T>
  : {
      [P in keyof T]-?: T[P] extends Array<infer U>
        ? Array<DeepRequired<U>>
        : T[P] extends ReadonlyArray<infer U2>
        ? DeepRequired<U2>
        : DeepRequired<T[P]>;
    };
and
  type ClientOptions = Required<Omit<Config, 'baseURLs'>> &
    DeepRequired<Pick<Config, 'baseURLs'>>;
```

Reply:
That seems like overkill. Maybe it's not important?

Reply:
meh, it's just typescript

Reply:
I don't think we should add it -- it's super-complex type code that future developers will have to maintain. This could probably be solved by an internal refactor -- we don't really need to expose these URLs.

I'd vote to leave it for now, and we can improve later.

Reply:
fine by me